cpp/src/Ice/ConnectionI.h

```diff
@@ -320,7 +308,6 @@ namespace Ice
         mutable Ice::ConnectionInfoPtr _info;
 
         ObjectAdapterPtr _adapter;
-        IceInternal::ServantManagerPtr _servantManager;
```

Comment:
It is not clear to me why we stored and passed around the servant manager. _servantManager is an immutable field of ObjectAdapter, so we could just ask ObjectAdapter whenever we needed it. That's what this PR does for dispatcher.